vital/types/local_cartesian.cxx

```diff
@@ -0,0 +1,446 @@
+        (Cos_LocalCart_Orientation * Cos_LocalCart_Origin_Lon + Sin_Lat_Sin_Orient * Sin_LocalCart_Origin_Lon) * v_MINUS_v0 +
+        (-Sin_LocalCart_Orientation * cos_lat_w_MINUS_w0);
+
+      Y = (-Sin_LocalCart_Orientation * Sin_LocalCart_Origin_Lon - Sin_Lat_Cos_Orient * Cos_LocalCart_Origin_Lon) * u_MINUS_u0 +
+        (Sin_LocalCart_Orientation * Cos_LocalCart_Origin_Lon - Sin_Lat_Cos_Orient * Sin_LocalCart_Origin_Lon) * v_MINUS_v0 +
+        (Cos_LocalCart_Orientation * cos_lat_w_MINUS_w0);
+
+      Z = (Cos_LocalCart_Origin_Lat * Cos_LocalCart_Origin_Lon) * u_MINUS_u0 +
+        (Cos_LocalCart_Origin_Lat * Sin_LocalCart_Origin_Lon) * v_MINUS_v0 +
+        Sin_LocalCart_Origin_Lat * w_MINUS_w0;
+    }
+
+    cartesian_coordinate << X, Y, Z;
+  }
+
+  /**
+   * @brief Converts cartesian coordinates to geocentric coordinates
+   *
+   * The function Convert_Local_Cartesian_To_Geocentric converts local cartesian
+   * coordinates (x, y, z) to geocentric coordinates (X, Y, Z) according to the
+   * current ellipsoid and local origin parameters.
+   *
+   * @param [in,out] cartesian_coordinate  : Local cartesian coordinate, in meters
+   * @param [in]     geocentric_coordinate : The geocentric location, in meters
+   */
+  void convert_to_geocentric(vector_3d const& cartesian_coordinate, vector_3d& geocentric_coordinate) const
+  {
+    double U, V, W;
+
+    double X = cartesian_coordinate.x();
+    double Y = cartesian_coordinate.y();
+    double Z = cartesian_coordinate.z();
+
+    if (LocalCart_Orientation == 0.0)
+    {
+      double sin_lat_y = Sin_LocalCart_Origin_Lat * Y;
+      double cos_lat_z = Cos_LocalCart_Origin_Lat * Z;
+
+      U = -Sin_LocalCart_Origin_Lon * X - sin_lat_y * Cos_LocalCart_Origin_Lon + cos_lat_z * Cos_LocalCart_Origin_Lon + u0;
+      V = Cos_LocalCart_Origin_Lon * X - sin_lat_y * Sin_LocalCart_Origin_Lon + cos_lat_z * Sin_LocalCart_Origin_Lon + v0;
+      W = Cos_LocalCart_Origin_Lat * Y + Sin_LocalCart_Origin_Lat * Z + w0;
+    }
+    else
+    {
+      double rotated_x, rotated_y;
+      double rotated_y_sin_lat, z_cos_lat;
+
+      rotated_x = Cos_LocalCart_Orientation * X + Sin_LocalCart_Orientation * Y;
+      rotated_y = -Sin_LocalCart_Orientation * X + Cos_LocalCart_Orientation * Y;
+
+      rotated_y_sin_lat = rotated_y * Sin_LocalCart_Origin_Lat;
+      z_cos_lat = Z * Cos_LocalCart_Origin_Lat;
+
+      U = -Sin_LocalCart_Origin_Lon * rotated_x - Cos_LocalCart_Origin_Lon * rotated_y_sin_lat + Cos_LocalCart_Origin_Lon * z_cos_lat + u0;
+      V = Cos_LocalCart_Origin_Lon * rotated_x - Sin_LocalCart_Origin_Lon * rotated_y_sin_lat + Sin_LocalCart_Origin_Lon * z_cos_lat + v0;
+      W = Cos_LocalCart_Origin_Lat * rotated_y + Sin_LocalCart_Origin_Lat * Z + w0;
+    }
+
+    geocentric_coordinate << U, V, W;
+  }
+private:
+  // Ellipsoid Parameters
+// All are set to a WGS 84 ellipsoid
```

Comment:
Why is this comment *not* indented? (BTW, it could probably be merged with the previous line...)